src/init.c

```diff
@@ -833,6 +833,20 @@ static void post_boot_hooks(void)
     }
 }
 
+volatile int ld_library_path_is_set;
+
+static int check_ld_library_path(void)
+{
+    if ( getenv("LD_LIBRARY_PATH") ) {
+        ld_library_path_is_set = 1;
+    }
+    else {
+        ld_library_path_is_set = 0;
+    }
+}
+
+check_ld_library_path();
```

Comment:
This doesn't do what you think it does. You can't just run functions at global scope in C. You need to call this from one of the init functions.